### ISSUE TITLE
Prepare data and develop joint embedding alignment for JEDI approach 

### ISSUE THREAD

**Issue by RobynMac25** [everycure-org/matrix#871]
**Aim:**
Start initial experimentation with JEDI (Joint Embedding Drug Intelligence) approach by preparing data and developing joint embedding alignment (without pipelining) 

**Description:**
- We aim to develop and validate the JEDI algorithm, which integrates textual embeddings from LLMs (e.g., PubMed BERT) with structural embeddings from GNNs (e.g., GraphSAGE) to predict novel drug-disease relationships
- This is a new and innovative approach and will therefore be highly experimental 
- As a first step, we will develop a joint embedding alignment methodology without pipelining 

**Tasks which need to be completed:**
- [ ] Prepare and preprocess embeddings from: LLMs trained on biomedical text (e.g., PubMed BERT) and GNN models trained on the knowledge graph (e.g., GraphSAGE)
- [ ] Align ground truth data for drug-disease relationships (positives and negatives)
- [ ] Implement contrastive learning to align textual and structural embeddings
- [ ] Define and optimize the contrastive loss function for embedding alignment

**This Sprint Goal is completed when:**
We have developed join embedding alignment for the JEDI approach in experimental form (working prototype, compatible with the Every Cure pipeline)
